modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,909 @@
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private Group getNewGroup() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccess")
+    public Object[][] dataToGetGroupSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+        String name = group.getDisplayName();
+
+        return new Object[][]{
+                {id, null, null, group},
+                {id, null, "members", group},
+                {id, name, null, group},
+        };
+    }
+
+    @Test(dataProvider = "dataForGetGroupSuccess")
+    public void testGetGroupSuccess(String id, String attributes, String excludeAttributes, Object objectGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group group = (Group) objectGroup;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getGroup(id, requiredAttributes)).thenReturn(group);
+        SCIMResponse outputScimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccessWithExcludeAttributes")
+    public Object[][] dataToGetGroupSuccessWithExcludeAttributes()
+            throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+        String name = group.getDisplayName();
+
+        return new Object[][]{
+                {id, null, "members", group},
+                {id, name, "members", group},
+        };
+    }
+
+    @Test(dataProvider = "dataForGetGroupSuccessWithExcludeAttributes")
+    public void testGetGroupSuccessWithExcludeAttributes(String id, String attributes,
```

Comment:
Since you have the same logic in testGetGroupSuccess test case as well, we can add assertions to check if the excluded attributes are not available in the response.

Reply:
Fixed in the commit 461abdefde0f2d32ab7fa7a3989ee89cec6d3834